spark/1.1.0.wso2v1/pom.xml

```diff
@@ -0,0 +1,368 @@
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>2.4.0</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            org.apache.spark.*,
+                        </Export-Package>
+                        <Private-Package>          <!-- TODO Export org.apache.hadoop.mapred | mapreduce ? -->
+                            org.apache.hadoop.*;-split-package:=merge-first,
+                            com.typesafe.config.*,
+                            org.j_paine.*;-split-package:=merge-first,
+                            akka.*,
+                            com.google.*,
+                            org.jets3t.*,
+                            org.json4s.*;-split-package:=merge-first,
+                            com.twitter.chill.*;-split-package:=merge-first,
+                            cern.*,hep.*,corejava.*,
+                            com.esotericsoftware.*;-split-package:=merge-first,
+                            org.netlib.*,
+                            org.jblas.*,
+                            breeze.*,
+                            spire.math.*,
+                            org.apache.commons.*;-split-package:=merge-first,
+                            net.jpountz.*,
+                            org.apache.mesos.*;-split-package:=merge-first,
+                            com.github.fommil.*,
+                        </Private-Package>
+                        <Import-Package>                           <!-- TODO Change import versions to version ranges -->
+                            org.jboss.netty.*;version="${org.jboss.netty.version}",
+                            io.netty.*,
+                            !scala.tools.*,
+                            !scala.reflect.api.*,
+                            scala.reflect.macros.internal.*,
+                            !scala.reflect.macros.*,
+                            !scala.quasiquotes.*,
+                            scala.*,
+                            org.eclipse.jetty.plus.*,
+                            org.eclipse.jetty.server.*,
+                            org.eclipse.jetty.util.*,
+                            org.eclipse.jetty.security.*,
+                            com.ning.compress.*,
+                            org.j_paine.*,
+                            com.codahale.metrics.*,
+                            com.fasterxml.jackson.*,
+                            org.apache.curator.*,
+                            org.apache.derby.*,
+                            org.apache.commons.lang3.*,
+                            org.xerial.snappy.*,
+                            org.objectweb.asm.*,
+                            *;resolution:=optional
```

Comment:
Import packages should have a version range. When importing packages the best practice is to use the semantic version approach with pattern like com.foo.bar.*;version=”[1.0.0, 1.1.0)”,
This informs that my bundle requires com.foo.bar.* package that is exported in the version range of >=1.0.0 and <1.1.0. This is normal way to ensure that there will be no API changes of the package from a minor version.